### PR TITLE
Adds column to display name of fansub in anime list

### PR DESCRIPTION
**Pull request by sunjayc**
This is designed how I personally like it, which means it will probably require some tweaking before being incorporated in the main branch.
Or you may even want to do something completely different.
Or not include it at all.
Just thought I'd put it up for review, though.

### CODE REVIEW

**Comment by erengy** [comment]
Here's a couple of things I could point out:

- You're storing the values in the anime database alongside metadata. Local data should be stored elsewhere. Currently other local data is stored in `settings.xml` (although this doesn't quite fit in there either).

- The methods in `anime_item.*` were ordered alphabetically, until you inserted `GetLastWatchedGroup()` randomly.

- Why is the text color changed to gray when the text is empty so there's nothing to draw?

- Instead of:

    ```cpp
    text = (std::wstring(L"[").append(anime_item.GetLastWatchedGroup()).append(L"]")).c_str();
    ```

    You could just:

    ```cpp
    text = L"[" + anime_item.GetLastWatchedGroup() + L"]";
    ```

    I don't think the brackets are necessary either, so it comes down to:

    ```cpp
    text = anime_item.GetLastWatchedGroup();
    ```

- You should be more elaborate at writing commit messages and ideally follow the same convention with the previous commits in the repository.

The main problem though, is that I can't really see this being used in a meaningful way. MAL's API doesn't allow setting the fansub group preference, Hummingbird doesn't have the field at all, so the data can't be synchronized. Why is it important to keep track of the last fansub group you watched for each series?

**Comment by sunjayc** [comment]
I like to read the release posts from fansub groups in a feed aggregator, and download my torrents that way. For me, it's helpful to know which group I'm getting which show from in an easily viewable way. It's probably not a very common use case and it's definitely a holdover from when I used to manage my list manually.

File: src/library/anime_item.cpp [hunk 1]
```diff
@@ -533,6 +533,10 @@ int Item::GetAvailableEpisodeCount() const {
   return static_cast<int>(local_info_.available_episodes.size());
 }
 
+const std::wstring& Item::GetLastWatchedGroup() const {
+  return local_info_.last_watched_group;
```
Comment (vendethiel) [inline src/library/anime_item.cpp]: why have a getter/setter here instead of a public field?

Comment (sunjayc) [inline src/library/anime_item.cpp]: A public field would probably be a better choice. I was attempting to mimic the surrounding code without (in hindsight) really thinking through it too hard.

Comment (erengy) [inline src/library/anime_item.cpp]: You did the right thing by keeping it consistent. Making it a public field would completely defeat the purpose of keeping series info, user info and local info as separate private member classes.

**Comment by sunjayc** [comment]
So should I fix this and resubmit or just close it?
I still would like to be able to see either last watched or preferred group in the main UI, at least as an optional field.

**Comment by CRS-ECHO51** [comment]
I think it's useless. We need external fansub database for displaying this.
Pada tanggal 8 Nov 2015 00.39, "sunjayc" <notifications@github.com> menulis:

**Comment by erengy** [comment]
@CRS-ECHO51 The idea here is to display the user's fansub group preferences, rather than the list of all fansub groups for each series. So, we don't need an external database for it.

I don't think that it's entirely useless, but after giving the idea some time, I still don't see it fit in with the way things are. The implementation may be different, but I'll consider adding fansub group preference as a column in the future.

**Pull request closed without merging**